src/components/Search/Search.tsx

```diff
@@ -1,30 +1,54 @@
-import { useState } from 'react'
-import { Button, FormControl, Input, Stack } from '@chakra-ui/react'
+import { useEffect, useState } from 'react'
 import { useGetMoviesBySearchWordQuery } from '../../services/moviesData'
+import { Button, FormControl, Input, Stack } from '@chakra-ui/react'
+import { useDebounce } from '@/hooks/useDebounce'
+import { SearchDropdown } from './SearchDropDown'
+import { Movie } from '@/types'
 
 export const Search = () => {
     const [query, setQuery] = useState('')
+    const [moviesList, setMovieList] = useState<Movie[]>([])
+    const [showSearchDropdown, setShowSearchDropdown] = useState<boolean>(false)
 
     // Get the data from the store
-    const { data } = useGetMoviesBySearchWordQuery(query)
-    const moviesBySearchWord = data?.results
+    const { data, isFetching: isFetchingMovies } =
+        // skip: prevents from making unnecessary calls when there is no query
+        useGetMoviesBySearchWordQuery(query, { skip: !query })
 
-    // const handleInput = debounce((e) => {
-    //     if (e.target.value.length > 2) {
-    //         setQuery(e.target.value)
-    //     }
-    // }, 1000)
+    const moviesBySearchWord: Movie[] = data?.results ?? []
 
-    // const handleSubmit = (e) => {
-    //     e.preventDefault()
-    // }
+    const handleEsckeyUp = (e: React.KeyboardEvent) => {
+        if (e.key === 'Escape') {
```

Comment:
Should use ` evt/event` because is more legible 
```suggestion
    const handleEsckeyUp = (evt: React.KeyboardEvent) => {
        if (evt.key === 'Escape') {
```